src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * Sets the command to be processed by the parser.
+     *
+     * @param command The user command to be processed.
+     */
+    public void setCommand(String command) {
+        this.command = command;
+    }
+
+    /**
+     * Processes the user command by executing the corresponding action.
+     * It handles commands like "bye," "list," "mark," "unmark," "delete," "todo," "deadline," and "event."
+     *
+     * @throws DukeException If there is an issue processing the command.
+     */
+    public void processCommand() throws DukeException {
+        String[] words = command.split("\\s+");
+
+        try {
+            ui.printDashes();
+            switch (words[0].toLowerCase()) {
+                case "bye":
+                    ui.showGoodbyeMessage();
+                    this.isRunning = false;
+                    break;
+                case "list":
+                    Ui.showTaskList(tasks);
+                    TaskList.getList(tasks);
+                    break;
+                case "mark":
+                    if (words.length > 1) {
+                        int index = Integer.parseInt(words[1]) - 1;
+                        if (index >= 0 && index < tasks.size()) {
+                            Task task = tasks.getTask(index);
+                            tasks.markStatus(task);
+                            storage.saveTasks(tasks);
+                            hasChanged = true; //flag, no need to change
+                        } else {
+                            Ui.showError("You have not created a task " + words[1] + " yet!");
+                        }
+                    }
+                    break;
+                case "unmark":
+                    if (words.length > 1) {
+                        int index = Integer.parseInt(words[1]) - 1;
+                        if (index >= 0 && index < tasks.size()) {
+                            Task task = tasks.getTask(index);
+                            tasks.unmarkStatus(task);
+                            storage.saveTasks(tasks);
+                            hasChanged = true; //flag, no need to change
+                        } else {
+                            Ui.showError("You have not created a task " + words[1] + " yet!");
+                        }
+                    }
+                    break;
+                case "delete":
+                    if (words.length > 1) {
+                        tasks.deleteTask(Integer.parseInt(words[1]) - 1);
+                        Ui.printNumberOfTasks(tasks);
+                        storage.saveTasks(tasks);
+                        hasChanged = true; //flag, no need to change
+                    }
+                    break;
+                case "todo":
+                    if (command.length() <= 5) {
+                        ui.showError("You gotta enter some task TO DO!");
+                    } else {
+                        tasks.addToDoTask(new ToDo(command.substring(5).trim()));
```

Comment:
Consider replacing any "magic numbers" (e.g., 5 and 3 in substring(5) and parts[3]) with named constants or variables with meaningful names to improve code readability.